### ISSUE TITLE
ALB-117: model_path checkpoint step leaks into fresh training

### ISSUE THREAD

**Issue by noahgift** [paiml/entrenar#273]
## Problem

When `model.path` in a training config points to a checkpoint directory containing
`model-step-N.apr` files, `load_transformer_model()` returns `checkpoint_step=N`.
The training loop then calls `set_initial_step(N)`, and if `max_steps < N`, training
exits immediately with `loss=0.000000` — no forward/backward steps are ever executed.

This manifests as:
```
Starting transformer training (CUDA GPU-resident)...
Reached max_steps=5, stopping training.
Total training time: 0.1s
Final loss: 0.000000
```

## Root Cause

`load_transformer_model()` has two sources for checkpoints:
1. **`output_dir`** — for training resume (step should be preserved)
2. **`model_path`** — for initial weight loading (step should be 0)

Both paths return `(model, checkpoint_step)`, but only `output_dir` represents
a genuine resume. When loading from `model_path`, the step embedded in the APR
filename (e.g., `model-step-14500.apr`) or metadata is carried through, causing
the LR schedule, bias correction, and max_steps check to behave incorrectly.

## Fix

Override checkpoint step to 0 when loading from `model_path` (not `output_dir`):
```rust
if let Some((model, _source_step)) = try_load_apr(model_path, config) {
    return Ok((model, 0));  // step=0: fresh training, not resume
}
```

## Discovery

Found while debugging loss=0.0 on NVIDIA GB10 (ALB-116 / aarch64 CUDA training).
The CPU test config (`pretrain-350m-cpu-test.yaml`) used `max_steps: 5` with
`model.path: "./checkpoints/albor-base-350m-v9"` which contains `model-step-14500.apr`.

## Contract

- **C-FRESH-TRAIN-001**: When `output_dir` has no checkpoint, `initial_step` MUST be 0
  regardless of the step encoded in `model_path`.

**Comment by noahgift** [comment]
Fixed in entrenar main (pushed 2026-03-14). Override checkpoint_step to 0 when loading from model_path (fresh training). Contract: C-FRESH-TRAIN-001 in albor contracts/.
